src/main/java/com/mixpanel/android/mpmetrics/MixpanelAPI.java

```diff
@@ -2109,6 +2111,84 @@ private static void checkIntentForInboundAppLink(Context context) {
         }
     }
 
+    public void trackGestures(Activity parent) {
+        parent.getWindow().getDecorView().findViewById(android.R.id.content).setOnTouchListener(new View.OnTouchListener() {
+
+            private void resetGesture() {
+                mFirstToSecondFingerDifference = -1;
+                mSecondFingerTimeDown = System.currentTimeMillis();
+                mGestureSteps = 0;
+                mBetweenSteps = -1;
+            }
+
+            @Override
+            public boolean onTouch(View v, MotionEvent event) {
+                if(event.getPointerCount() > 2) {
+                    resetGesture();
+                    return false;
+                }
+
+                if(event.getActionMasked() == MotionEvent.ACTION_DOWN) {
+                    mFirstToSecondFingerDifference = System.currentTimeMillis();
+                }
+
+                if(event.getActionMasked() == MotionEvent.ACTION_POINTER_DOWN && (System.currentTimeMillis() - mFirstToSecondFingerDifference) < 100) {
+                    mSecondFingerTimeDown = System.currentTimeMillis();
+                    mFirstToSecondFingerDifference = 100;
+                    if (System.currentTimeMillis() - mBetweenSteps > 1000 && mBetweenSteps != -1) {
+                        resetGesture();
+                        return false;
+                    }
+
+                } else if (event.getActionMasked() == MotionEvent.ACTION_POINTER_DOWN) {
+                    resetGesture();
+                    return false;
+                }
+
+                if(event.getActionMasked() == MotionEvent.ACTION_POINTER_UP && mFirstToSecondFingerDifference == 100) {
+                    mFirstToSecondFingerDifference = System.currentTimeMillis();
+                } else if (event.getActionMasked() == MotionEvent.ACTION_POINTER_UP) {
+                    resetGesture();
+                    return false;
+                }
+
+                if(event.getActionMasked() == MotionEvent.ACTION_UP && (System.currentTimeMillis() - mFirstToSecondFingerDifference) < 100 && mFirstToSecondFingerDifference > 0){
```

Comment:
I don't see why you need to check `mFirstToSecondFingerDifference > 0` if you previously check `System.currentTimeMillis() - mFirstToSecondFingerDifference) < 100`.